apps/webapp/web/router.ex

```diff
@@ -36,7 +36,8 @@ defmodule Webapp.Router do
     get "/login", PageController, :login
     get "/docs", PageController, :docs
     get "/docs/index", DocController, :index
-    get "/docs/:filename", DocController, :show
+    get "/docs/:filename", PageController, :docs
```

Comment:
Please revert this route back to `get "/docs/:filename", DocController, :show`.  Then in the `DocController` use pattern matching to render **index.html** or the **json** response.

For example, in the controller you could try something like:

```elixir
def show(conn, %{"format" => "html", "filename" => filename} do
  render conn, Webapp.PageView, "index.html"
end

def show(conn, %{"format" => "json", "filename" => filename} do
  # do your json rendering of the file
end
```

In order for the above thing to work you'll need to accept "json" format in the browser pipeline in the router.

```elixir
  plug :accepts, ["html", "json"]
```

Then when you request json, you'll just need to add `_format=json` to request's the query string.